src/format-importer.ts

```diff
@@ -139,6 +139,45 @@ export abstract class FormatImporter {
 		return null;
 	}
 
+	/**
+	 * Resolves a unique path for the attachment file being saved.
+	 * Ensures that the parent directory exists and dedupes the
+	 * filename if the destination filename already exists.
+	 *
+	 * NOTE: This is a duplicate of `fileManager.getAvaialbePathForAttachment`
+	 * which adds two key adjustments to aid Importer:
+	 *   - Use the provided `sourcePath` even if the file doesn't exist yet.
+	 *   - Avoid duplicating a list of provided filesnames that do not yet exist, but will in the future.
+	 *
+	 * @param filename Name of the attachment being saved
+	 * @param claimedPaths List of filepaths that may not exist yet but will in the future.
+	 * @returns Full path for where the attachment should be saved, according to the user's settings
+	 */
+	async getAvailablePathForAttachment(filename: string, claimedPaths: string[]): Promise<string> {
+		const outputFolder = await this.getOutputFolder();
```

Comment:
Might be better if this is cached somehow?

Reply:
Added class-level caching to `getOutputFolder`.